rust/src/dns/dns.rs

```diff
@@ -944,6 +944,121 @@ pub unsafe extern "C" fn SCDnsTxGetAnswerName(
     false
 }
 
+/// Get the DNS response authority name at index i.
+#[no_mangle]
+pub unsafe extern "C" fn SCDnsTxGetAuthorityName(
+    tx: &mut DNSTransaction, i: u32, buf: *mut *const u8, len: *mut u32,
+) -> bool { 
+    let index = i as usize;
+
+    if let Some(response) = &tx.response {
+        if let Some(record) = response.authorities.get(index) {
+            if !record.name.is_empty() {
+                *buf = record.name.as_ptr();
+                *len = record.name.len() as u32;
+                return true;
+            }
+        }
+    }
+
+    false
+}
+
+/// Get the DNS response additional name at index i.
+#[no_mangle]
+pub unsafe extern "C" fn SCDnsTxGetAdditionalName(
+    tx: &mut DNSTransaction, i: u32, buf: *mut *const u8, len: *mut u32,
+) -> bool { 
+    let index = i as usize;
+
+    if let Some(response) = &tx.response {
+        if let Some(record) = response.additionals.get(index) {
+            if !record.name.is_empty() {
+                *buf = record.name.as_ptr();
+                *len = record.name.len() as u32;
+                return true;
+            }
+        }
+    }
+
+    false
+}
+
+#[inline]
+unsafe fn dns_get_record_rdata(data: &DNSRData, buf: *mut *const u8, len: *mut u32) -> bool {
```

Comment:
Does the whole function have to be defined as `unsafe`? usually for our Rust side code, I see we try to restringe the `unsafe` scope as much as possible.

(nit) We usually separate the functions that are for the C API from the ones that are for the Rust side (so this could move before we start defining the C API)

Reply:
Agreed.